ui/src/features/Editor/components/OverlayUI/components/DebugPanel/index.tsx

```diff
@@ -0,0 +1,218 @@
+import {
+  CaretDownIcon,
+  CaretUpIcon,
+  CodeIcon,
+  CornersInIcon,
+  CornersOutIcon,
+  EyeIcon,
+  MinusIcon,
+  TerminalIcon,
+} from "@phosphor-icons/react";
+import { memo, useEffect, useRef, useState } from "react";
+
+import {
+  ResizableHandle,
+  ResizablePanel,
+  ResizablePanelGroup,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  Tabs,
+  TabsContent,
+  TabsList,
+  TabsTrigger,
+} from "@flow/components";
+import { useT } from "@flow/lib/i18n";
+
+import DebugLogs from "./DebugLogs";
+import DebugPreview from "./DebugPreview";
+import { DataTable } from "./DebugPreview/components";
+import useHooks from "./hooks";
+
+type Props = {
+  fullscreenDebug: boolean;
+  onFullScreenExpand: () => void;
+};
+
+const DebugPanel: React.FC<Props> = ({
+  fullscreenDebug,
+  onFullScreenExpand,
+}) => {
+  const {
+    debugJobId,
+    debugJobState,
+    fileType,
+    expanded,
+    minimized,
+    showTempPossibleIssuesDialog,
+    selectedDataURL,
+    dataURLs,
+    selectedOutputData,
+    isLoadingData,
+    handleExpand,
+    handleMinimize,
+    handleTabChange,
+    handleShowTempPossibleIssuesDialogClose,
+    handleSelectedDataChange,
+  } = useHooks();
+  const t = useT();
+  const [tabValue, setTabValue] = useState("debug-logs");
+  const hasSwitchedToViewerRef = useRef(false);
+  const debugJobIdRef = useRef(debugJobId);
+
+  useEffect(() => {
+    if (debugJobId !== debugJobIdRef.current) {
+      debugJobIdRef.current = debugJobId;
+    }
+
+    if (dataURLs && !hasSwitchedToViewerRef.current) {
+      setTabValue("debug-viewer");
+      hasSwitchedToViewerRef.current = true;
+    }
+
+    if (!dataURLs) {
+      setTabValue("debug-logs");
+      hasSwitchedToViewerRef.current = false;
+    }
+  }, [dataURLs, debugJobId]);
```

Comment:
Be very careful with doing too much in a single useEffect. Easier to read and safer typically to just have separate useEffects for each thing you want done.  In this case, if you split the first if statement into its own, we would easily be able to understand it'll only ever re-fire if debugJobId changes. Currently, we have to be cautious and think about undesired side-effects to debugJobIdRef if the dataURLs changes, something that is completely unrelated. Make sense?